code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -80,9 +80,12 @@
 			user << "<span class='danger'>[src] buzzes and beeps.</span>"
 
 /mob/living/simple_animal/bot/cleanbot/process_scan(obj/effect/decal/cleanable/D)
+	. = list()
 	for(var/T in target_types)
 		if(istype(D, T))
-			return D
+			. += D
+	if(.)

```

Comment:
bad, ```.``` can be an empty list in this case, ```if(.)``` is always true because you set it to a list object above.
you also can't check the ```.len``` of it because it's not a list, a way around this would be ```if(.[1])``` to check that ```.``` atleast contains 1 element.